sources/api/apiclient/src/main.rs

```diff
@@ -138,21 +215,183 @@ fn parse_raw_args(args: Vec<String>) -> Subcommand {
+    let mut subcommand = None;
+    let mut subcommand_args = Vec::new();
+
+    let mut iter = args.into_iter();
+    while let Some(arg) = iter.next() {
+        match arg.as_ref() {
+            // Subcommands
+            "check" | "apply" | "cancel" if subcommand.is_none() && !arg.starts_with('-') => {
+                subcommand = Some(arg)
+            }
+
+            // Other arguments are passed to the subcommand parser
+            _ => subcommand_args.push(arg),
+        }
+    }
+
+    let update = match subcommand.as_deref() {
+        Some("check") => parse_check_args(subcommand_args),
+        Some("apply") => parse_apply_args(subcommand_args),
+        Some("cancel") => parse_cancel_args(subcommand_args),
+        _ => usage_msg("Missing or unknown subcommand for 'update'"),
+    };
+
+    Subcommand::Update(update)
+}
+
+/// Parses arguments for the 'update check' subcommand.
+fn parse_check_args(args: Vec<String>) -> UpdateSubcommand {
+    if !args.is_empty() {
+        usage_msg(&format!("Unknown arguments: {}", args.join(", ")));
+    }
+    UpdateSubcommand::Check(CheckArgs {})
+}
+
+/// Parses arguments for the 'update apply' subcommand.
+fn parse_apply_args(args: Vec<String>) -> UpdateSubcommand {
+    let mut check = false;
+    let mut reboot = false;
+
+    let mut iter = args.into_iter();
+    while let Some(arg) = iter.next() {
+        match arg.as_ref() {
+            "-c" | "--check" => check = true,
+            "-r" | "--reboot" => reboot = true,
+
+            x => usage_msg(&format!("Unknown argument '{}'", x)),
+        }
+    }
+
+    UpdateSubcommand::Apply(ApplyArgs { check, reboot })
+}
+
+/// Parses arguments for the 'update cancel' subcommand.
+fn parse_cancel_args(args: Vec<String>) -> UpdateSubcommand {
+    if !args.is_empty() {
+        usage_msg(&format!("Unknown arguments: {}", args.join(", ")));
+    }
+    UpdateSubcommand::Cancel(CancelArgs {})
+}
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// Requests a reboot through the API.
+async fn reboot(args: &Args) -> Result<()> {
+    let uri = "/actions/reboot";
+    let method = "POST";
+    let (_status, _body) = apiclient::raw_request(&args.socket_path, uri, method, None)
+        .await
+        .context(error::Request { uri, method })?;
+
+    info!("Rebooting, goodbye...");
+    Ok(())
+}
+
+/// Requests an update status check through the API, printing the updated status, in a pretty
+/// format if possible.
+async fn check(args: &Args) -> Result<String> {
+    let output = update::check(&args.socket_path)
+        .await
+        .context(error::Check)?;
+
+    match serde_json::from_str::<serde_json::Value>(&output)
+        .and_then(|v| serde_json::to_string_pretty(&v))
+    {
+        Ok(pretty) => println!("{}", pretty),
+        Err(e) => {
+            warn!("Unable to pretty-print response (invalid JSON?): {}", e);
+            println!("{}", output);
+        }
+    }
```

Comment:
nit: 
It seems like serde_json::value::Value implements `Display` with a pretty print format: https://docs.serde.rs/serde_json/value/enum.Value.html#impl-Display

So maybe we can simplify this to
```suggestion
    match serde_json::from_str::<serde_json::Value>(&output)
    {
        Ok(value) => println!("{:#}", value),
        Err(e) => {
            warn!("Unable to deserialize response (invalid JSON?): {}", e);
            println!("{}", output);
        }
    }
```